src/main/java/com/eu/habbo/habbohotel/users/HabboItem.java

```diff
@@ -556,4 +560,61 @@ public Rectangle getRectangle(int marginX, int marginY) {
                 this.getBaseItem().getLength() + (marginY * 2),
                 this.getRotation());
     }
+
+    public boolean isItemHiddenByAreaHider() {
+
+        if (this instanceof InteractionAreaHider) {
+            return false;
+        }
+
+        Room room = Emulator.getGameEnvironment().getRoomManager().getRoom(this.getRoomId());
+        List<HabboItem> areaHiders = room.getAreaHiders(true);
+        Gson gson = new Gson();
+
+        short x = this.getX();
+        short y = this.getY();
+
+        if (this.getBaseItem().getType() == FurnitureType.WALL) {
+            x += (short) (this.getRotation() == 0 ? 1 : 0);
+            y += (short) (this.getRotation() == 1 ? 1 : 0);
+        }
+
+        RoomTile tile = room.getLayout().getTile(x, y);
+
+        for (HabboItem areaHider : areaHiders) {
+            String extraData = areaHider.getExtradata();
+
+            if (extraData != null && extraData.trim().startsWith("{")) {
+                try {
+                    InteractionAreaHider.JsonData data = gson.fromJson(extraData, InteractionAreaHider.JsonData.class);
+
+                    Rectangle rectangle = new Rectangle(data.rootX, data.rootY, data.width, data.length);
+                    boolean isInRectangle = RoomLayout.tileInSquare(rectangle, tile);
+
+                    if (data.invertEnabled != isInRectangle) {
+                        return true;
+                    }
+
+                } catch (JsonSyntaxException e) {
+                    LOGGER.error("Error parsing extradata for AreaHider ({}): {}", areaHider.getId(), e.getMessage());
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public void parseWallItemPosition(String wallPosition, HabboItem item) {
```

Comment:
This can be a static method since the `HabboItem item` is passed as a parameter. `parse` is also a wrong description because it's also updating the item, `update...` should be better.

Also this method probably needs an `if` check to verify that this is indeed a wall item.